src-tauri/src/settings.rs

```diff
@@ -49,20 +52,29 @@ pub fn create_permissions() {
     let _ = fs::create_dir(get_magnus_data_dir_path());
 
     // create the permissions.json file with all false values
-    let _ = fs::write(get_file_path(), pretty_json.as_bytes()).expect("Failed to update permissions.json!");
+    let _ = fs::write(get_permissions_file_path(), pretty_json.as_bytes()).expect("Failed to update permissions.json!");
 }
 
 pub fn update_permissions(permissions: Value) {
-    let pretty_json = to_string_pretty(&permissions).unwrap();
-    let _ = fs::write(get_file_path(), pretty_json.as_bytes()).expect("Failed to update permissions.json!");
+    if permissions != Value::Object(Map::new()) {
+        let pretty_json = to_string_pretty(&permissions).unwrap();
+        let _ = fs::write(get_permissions_file_path(), pretty_json.as_bytes()).expect("Failed to update permissions.json!");
+    }
 }
 
 pub fn get_permissions() -> Value {
-    match File::open(get_file_path()) {
+    match File::open(get_permissions_file_path()) {
         Ok(mut file) => {
             let mut json_string = String::new();
             file.read_to_string(&mut json_string).expect("Failed to read permissions.json!");
             let permissions: Value = serde_json::from_str(&json_string).expect("Failed to parse permissions.json!");
+
+            // fix empty permisisons file
+            if permissions == Value::Object(Map::new()) {
```

Comment:
or maybe check the length of it?

Reply:
we essentially are checking if its equal to an empty object ("{}")

Reply:
This is really nit picky so feel free to address this how you feel fit. Since we are technically creating a new Map Object here each time it seems like a waste of resources in a way. Ultimately it probably wont matter at all lol, but just a thought. Im cool with whatever you decide to do.